testing/mongo-c-driver/APKBUILD

```diff
@@ -0,0 +1,58 @@
+# Maintainer: Leonardo Arena <rnalrd@alpinelinux.org>
+pkgname="mongo-c-driver"
+pkgver="1.14.0"
+pkgrel=0
+pkgdesc="Client library written in C for MongoDB"
+url="https://github.com/mongodb/mongo-c-driver"
+arch="all"
+license="Apache-2.0"
+depends_dev="${pkgname} cyrus-sasl-dev icu-dev openssl-dev" 
+makedepends="${depends_dev} cmake py3-sphinx"
+subpackages="${pkgname}-libs-static ${pkgname}-libs ${pkgname}-dev ${pkgname}-doc"
+source="${pkgname}-${pkgver}.tar.gz::https://github.com/mongodb/${pkgname}/archive/${pkgver}.tar.gz"
+
+build() {
+	cmake \
+		-DCMAKE_BUILD_TYPE=Release \
+		-DCMAKE_INSTALL_PREFIX:PATH=/usr \
+		-DCMAKE_INSTALL_LIBDIR:PATH=lib \
+		-DENABLE_BSON:STRING=ON \
+		-DENABLE_MONGOC:BOOL=ON \
+		-DENABLE_SSL:STRING=OPENSSL \
+		-DENABLE_AUTOMATIC_INIT_AND_CLEANUP:BOOL=OFF \
+		-DENABLE_MAN_PAGES:BOOL=ON \
+		-DENABLE_TESTS:BOOL=ON \
+		-DENABLE_EXAMPLES:BOOL=OFF \
+		-DSPHINX_EXECUTABLE:STRING=/usr/bin/sphinx-build-3 \
+		-DCMAKE_SKIP_RPATH:BOOL=ON \
+		-DENABLE_SHM_COUNTERS:BOOL=ON \
+		-DENABLE_SASL:STRING=CYRUS \
+		-DENABLE_ICU:STRING=ON \
+		-DENABLE_CRYPTO_SYSTEM_PROFILE:BOOL=ON \
+		-DENABLE_UNINSTALL:BOOL=OFF \
+		-DBUILD_VERSION:STRING=${pkgver} \
+		.
+	make
+}
+
+check() {
+	export MONGOC_TEST_SKIP_MOCK=on
+	export MONGOC_TEST_SKIP_SLOW=on
+	export MONGOC_TEST_SKIP_LIVE=on
+	make check
+}
+
+package() {
+	make DESTDIR="${pkgdir}" install
+	mkdir "${pkgdir}/usr/share/doc"
+	mv "${pkgdir}/usr/share/${pkgname}" "${pkgdir}/usr/share/doc"
```

Comment:
@tcely, after this command `/usr/share` folder still exist. But `mongo-c-driver-1.14.0-r0.apk` package not contains this folder.
Is this folder automatically deleted from package if empty?
May be need add similar check for `/usr/lib` folder into `abuild` script?

Reply:
See https://github.com/alpinelinux/abuild/blob/297de93aef3110f6ced8926e64e4204692b3b74e/abuild.in#L1670

Even if `default_static` or `default_libs` had a similar command, `/usr/lib` is not empty until the `dev` split function runs.

Reply:
Think one example will be more easy to describe my question.
Say exist script:
```sh
pkgname="test"
pkgver="0.0.1"
pkgrel=0
pkgdesc="test"
url="https://example.com"
arch="noarch"
license="Apache-2.0"
subpackages="${pkgname}-doc"

build() {
	true
}

check() {
	true
}

package() {
	mkdir -p "${pkgdir}/usr/share/doc" \
	         "${pkgdir}/usr/lib"
	touch "${pkgdir}/usr/share/doc/test"
}
```
Script creates two packages and `test-0.0.1-r0.apk`  contains empty `/usr/lib` folder.

To fix it I created https://github.com/alpinelinux/abuild/pull/94

Is PR ready to merge?

Reply:
Something like that function could be used. I'd suggest using `find` to generate the list of directories for `rmdir` instead of a static list.

Reply:
Application may package own empty directories.
If `find` utility is used for empty folder deletion, then package may be broken.
More safe delete only directories used by `abuild`.